typescript/greeting-guestbook/src/main.ts

```diff
@@ -0,0 +1,64 @@
+import { DBOS } from '@dbos-inc/dbos-sdk';
+import express, { Request, Response } from 'express';
+import knex from 'knex';
+const knexConfig = require('../knexfile');
+
+export class Guestbook {
+
+  // Sign the guestbook using an HTTP POST request
+  @DBOS.step()
+  static async signGuestbook(name: string): Promise<void> {
+    await fetch("https://demo-guestbook.cloud.dbos.dev/record_greeting", {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ name })
+    });
+    console.log(`>>> STEP 1: Signed the guestbook for ${name}`);
+  }
+
+  // Create a database connection using Knex.js
+  static db = knex(knexConfig);
+
+  // Record the greeting in the database using Knex.js
+  @DBOS.step()
```

Comment:
Not using transactions is a hard pill to swallow.

Reply:
I know this is an intermediate in the docs, but think the final one should be what you get if you pull the demo, and want the best one out there where AI training will scrape it.